### DISCUSSION TITLE
fetch data from rest api and insert into db every x period

### DISCUSSION THREAD

**Discussion by Trigie** [Budibase/budibase#3746] [Help]
Hello,

I'm new to budibase, I want to fetch some data from an external data source every x periode and then insert it into a database.
Retrieving the external data works but i'm not able to insert the data by using the automation feature.
The output of the external query is ok and contains multiple rows. 
How can I insert these in the local Budibase table? I tried this seup:

![image](https://user-images.githubusercontent.com/33255473/145427109-55d7b9ab-606d-47f6-9074-68775ec629a8.png)
This is the error:
![image](https://user-images.githubusercontent.com/33255473/145426736-23e89659-9308-4abf-9b62-fee97cdad0af.png)

**Comment by mjashanks** [comment]
Hi @Trigie - welcome!

Unfortunately, it's not possible to insert multiple rows at once, although we would very much like to add this feature.

What you might be able to do is run the CRON trigger very frequently, so that you are able to insert one at a time. You'd probably need to                        find the first item in your REST response that wasn't already imported. You could use the Javascript action to do this.

Definitely not an ideal solution - but might work.

**Reply by Trigie** [reply]
Ok thanks for the quick response, this specific query will always return multiple rows so this aint an option...
I found another query with only one row (with less columns) but still get the Pouch error.

Is this the way to use the results of the fetch from the external data connector?

{{ steps.1.rows.0.id }}

**Reply by mjashanks** [reply]
Yes - that how to use it. We are deploying a fix for that pouch error today.
